rfcs/proposed/range_algorithms_3/README.md

```diff
@@ -0,0 +1,48 @@
+# Support the third portion of the oneDPL Range APIs
+
+## Introduction
+Based on statistics (observing C++ code within github.com) for the usage of popular algorithms, the following
+range-based APIs are suggested to be implemented next in oneDPL.
+`swap_ranges`, `reverse`, `reverse_copy`, `unique`, `unique_copy`, `set_intersection`, `set_union`, `set_difference`, `set_symmetric_difference`,
+`includes`, `nth_element`, `partition`, `partition_copy`, `remove_copy`, `remove_copy_if`, `lexicographical_compare`, `rotate`, `rotate_copy`,
+`uninitialized_copy`, `uninitialized_move`, `uninitialized_fill`, `uninitialized_default_construct`, `uninitialized_value_construct`, `destroy`
+
+## Motivations
+The feature is proposed as the next step of range-based API support for oneDPL.
+
+### Key Requirements
+- The range-based signatures for the mentioned API should correspond to the [proposed specification](https://github.com/uxlfoundation/oneAPI-spec/pull/614)
+that is based on the [C++ standardization proposal P3179](https://wg21.link/p3179).
+- The proposed implementation should support all oneDPL execution policies: `seq`, `unseq`, `par`, `par_unseq`, and `device_policy`.
+- For the algprithms `reverse_copy`, `rotate_copy` return values semantic might be changed in the future or this algorithms will be not added to oneDPL.
+
+### Implementation proposal
+- The implementation is supposed to rely on existing range-based or iterator-based algorithm patterns, which are already
+implemented in oneDPL.
+- Several algorithms described in P3179 have slightly different semantics. To implement these, some existing algorithm patterns
+might require modifications or new versions.
+- For the algorithms, which work with uninitialized memory (`uninitialized_*` and `destroy` form `<memory>`) an additional require is proposed: `oneapi:dpl::value_adressable`.
+The require can be implemented via C++ concept: `template<typename R> concept value_adressable = requires { std::addressof(*std::ranges::iterator_t<R>{}); };`
```

Comment:
It would be good to explain potential benefits of this idea.

Reply:
yes, it looks that `no-throw-input-iterator` is sufficient.